### DISCUSSION TITLE
The AKS primality test

### DISCUSSION THREAD

**Discussion by valeriob01** [preda/gpuowl#211] [Ideas]
Wikipedia: https://en.wikipedia.org/wiki/AKS_primality_test

Github: https://github.com/tingliu/aks-primality-test

**Comment by valeriob01** [comment]
And various other algorithms are proposed on Wikipedia for primality tests:

https://en.wikipedia.org/wiki/Adleman%E2%80%93Pomerance%E2%80%93Rumely_primality_test

https://en.wikipedia.org/wiki/Elliptic_curve_primality

https://en.wikipedia.org/wiki/Baillie%E2%80%93PSW_primality_test

**Comment by preda** [comment]
For special form numbers such as mersennes (2^p - 1), both LL and PRP are faster than these general primality tests (that work on any numbers).

The complexity of LL and PRP is O(p^2 * log(p)) which is O(log(Mp)^2 * loglog(Mp)).

**Reply by valeriob01** [reply]
> For special form numbers such as mersennes (2^p - 1), both LL and PRP are faster than these general primality tests (that work on any numbers).
> 
> The complexity of LL and PRP is O(p^2 * log(p)) which is O(log(Mp)^2 * loglog(Mp)).

You should write about PRP on Wikipedia then.
